src/solver/optimisation/optim_post_process_list.cpp

```diff
@@ -0,0 +1,44 @@
+#include "optim_post_process_list.h"
+#include "post_process_commands.h"
+
+namespace Antares::Solver::Simulation
+{
+
+OptPostProcessList::OptPostProcessList(PROBLEME_HEBDO* problemeHebdo,
+                                       uint thread_number,
+                                       AreaList& areas,
+                                       SheddingPolicy sheddingPolicy,
+                                       SimplexOptimization splxOptimization,
+                                       Calendar& calendar)
+
+    : interfacePostProcessList(problemeHebdo, thread_number)
+{
+    post_process_list.push_back(std::make_unique<DispatchableMarginPostProcessCmd>(
+        problemeHebdo_, 
+        thread_number_, 
+        areas));
+    post_process_list.push_back(std::make_unique<HydroLevelsUpdatePostProcessCmd>(
+        problemeHebdo_, 
+        areas, 
+        false, 
+        false));
+    post_process_list.push_back(std::make_unique<RemixHydroPostProcessCmd>(
+        problemeHebdo_,
+        areas,
+        sheddingPolicy,
+        splxOptimization,
+        thread_number));
+    post_process_list.push_back(std::make_unique<HydroLevelsUpdatePostProcessCmd>(
+        problemeHebdo_,
+        areas,
+        true,
+        false));
+    post_process_list.push_back(std::make_unique<InterpolateWaterValuePostProcessCmd>(
+        problemeHebdo_,
+        areas,
+        calendar));
```

Comment:
I know you don't like if..else much, but IMO it would be clearer to have
```cpp
post_process_list.push_back(...);
if (adqPatch)
    post_process_list.push_back(std::make_unique<DTGmarginForAdqPatchPostProcessCmd>(...);
post_process_list.push_back(...);
```
It would avoid code duplication.

Reply:
You could hide this in a factory function
```cpp
postProcessListFactory(bool adqPatch) {
// ...
}
```

Reply:
This should be discussed more deeply. 
Before taking a decision, let's look at **adequacy.cpp**, which has not been touched yet.